ugvc/pipelines/sv_stats_collect.py

```diff
@@ -0,0 +1,246 @@
+            df_base["svlen"].abs(),
+            bins=[0, 100, 300, 500, 1000, 5000, 10000, 100000, 1000000, float("inf")],
+            labels=["50-100", "100-300", "300-500", "0.5-1k", "1k-5k", "5k-10k", "10k-100k", "100k-1M", ">1M"],
+            right=False,
+        )
+        df_calls["binned_svlens"] = pd.cut(
+            df_calls["svlen"].abs(),
+            bins=[0, 100, 300, 500, 1000, 5000, 10000, 100000, 1000000, float("inf")],
+            labels=["50-100", "100-300", "300-500", "0.5-1k", "1k-5k", "5k-10k", "10k-100k", "100k-1M", ">1M"],
+            right=False,
+        )
+        # Collect concordance statistics
+
+        category = ["ALL", "DEL", "DUP", "INV", "INS", "CTX"]
+        for c in category:
+            if c == "ALL":
+                df_base_c = df_base
+                df_calls_c = df_calls
+            else:
+                df_base_c = df_base.query(f"svtype == '{c}'")
+                df_calls_c = df_calls.query(f"svtype == '{c}'")
+
+            concordance_stats[f"{c}_concordance"] = concordance_with_gt(df_base_c, df_calls_c)  # type: ignore
+            concordance_stats[f"{c}_roc"] = concordance_with_gt_roc(df_base_c, df_calls_c)  # type: ignore
+
+        category = ["ALL", "DEL", "INS"]
+        bins = ["50-100", "100-300", "300-500", "0.5-1k", "1k-5k", "5k-10k", "10k-100k", "100k-1M", ">1M"]
+        for c in category:
+            for len_bin in bins:
+                if c == "ALL":
+                    df_base_c = df_base
+                    df_calls_c = df_calls
+                else:
+                    df_base_c = df_base.query(f"svtype == '{c}'")
+                    df_calls_c = df_calls.query(f"svtype == '{c}'")
+                df_base_c = df_base_c.query(f"binned_svlens == '{len_bin}'")
+                df_calls_c = df_calls_c.query(f"binned_svlens == '{len_bin}'")
+                concordance_stats[f"{c}_{len_bin}_concordance"] = concordance_with_gt(df_base_c, df_calls_c).drop(
+                    ["FP", "Precision", "F1"]
+                )
+        fp_stats = (
+            df_calls.query("label=='FP'")[["svtype", "binned_svlens"]].value_counts().sort_index().astype("int64")
+        )
+
+    return sv_stats, concordance_stats, fp_stats
+
+
+def run(args: list[str]):
+    """
+    Run the SV statistics collection pipeline.
+
+    Parameters
+    ----------
+    args : list[str]
+        Command line arguments.
+    """
+    parser = argparse.ArgumentParser(description="Collect SV statistics from a VCF file.")
+    parser.add_argument("svcall_vcf", type=str, help="Path to the SV call VCF file.")
+    parser.add_argument("output_file", type=str, help="Output PKL file.")
+    parser.add_argument(
+        "--concordance_h5", type=str, help="Path to the concordance HDF5 file.", default=None, required=False
+    )
+
+    p_args = parser.parse_args(args[1:])
+
+    sv_stats, concordance_stats, fp_stats_df = collect_sv_stats(p_args.svcall_vcf, p_args.concordance_h5)
+    results = {}
```

Comment:
maybe this is clear, but maybe add how do we get such a concordance file? Or its required fields/ format